### PR TITLE
Closes #69 | Implement XStoryCloze Dataloader

### LINKED ISSUE #69: Create dataset loader for XStoryCloze
Dataloader name: `xstorycloze/xstorycloze.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?xstorycloze

|     Dataset| xstorycloze |
|-------------|---|
| Description | XStoryCloze consists of the professionally translated version of the English StoryCloze dataset (Spring 2016 version) to 10 non-English languages. This dataset is released by Meta AI. |
| Subsets | id, my |
| Languages | ind, mya |
| Tasks | Commonsense Reasoning |
| License | Creative Commons Attribution Share Alike 4.0 (cc-by-sa-4.0) |
| Homepage | https://huggingface.co/datasets/juletxara/xstory_cloze |
| HF URL | https://huggingface.co/datasets/juletxara/xstory_cloze |
| Paper URL | https://aclanthology.org/2022.emnlp-main.616 |

### PR DESCRIPTION
**Pull request by akhdanfadh**
Closes #69.

I implemented one config per language/subset. Thus, configs will look like this: `xstorycloze_id_source`, `xstorycloze_id_seacrowd_ssp`, or `xstorycloze_id_seacrowd_qa`. When testing, pass `xstorycloze_<subset>` to the `--subset_id` parameter.

Note that I do not follow the specified tasks in #69 since this dataset is more suitable for `Tasks.COMMONSENSE_REASONING` instead of `Tasks.QUESTION_ANSWERING` even though they both share schema.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/xstorycloze/xstorycloze.py
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/xstorycloze/xstorycloze.py:111]: Small nit: because you'll have two schemas, it might be better to be explicit here and just write the schema name itself. Same comment in the `_generate_examples` code.

**Review by ljvmiranda921 — COMMENTED** [review]
Data loader works as intended! Thanks for the PR @akhdanfadh . Just wondering why there's a self-supervised learning schema implementation? The [Hugging Face homepage](https://huggingface.co/datasets/juletxara/xstory_cloze#supported-tasks-and-leaderboards) only treats this as a common-sense reasoning task?

I don't mind removing the `_ssp` schema implementation though.

**Comment by sabilmakbar** [comment]
> Data loader works as intended! Thanks for the PR @akhdanfadh . Just wondering why there's a self-supervised learning schema implementation? The [Hugging Face homepage](https://huggingface.co/datasets/juletxara/xstory_cloze#supported-tasks-and-leaderboards) only treats this as a common-sense reasoning task?
> 
> I don't mind removing the `_ssp` schema implementation though.

I do agree w/ LJ's comment on this. As long as a more valuable task can be performed, we can leave the `self-supervised pertaining` task unimplemented. I think last time we had an agreement on how [`ssp` task](https://discord.com/channels/1164061531227557920/1178711836409991348/1178895027020251166) is being implemented, right?

File: seacrowd/sea_datasets/xstorycloze/xstorycloze.py
Comment (akhdanfadh) [reply seacrowd/sea_datasets/xstorycloze/xstorycloze.py:111]: Hi sorry for the late reply, I implemented `_ssp` here because in the dataloader issue #69 there are two tasks and one for language modelling.

Comment (sabilmakbar) [reply seacrowd/sea_datasets/xstorycloze/xstorycloze.py:111]: Okay then, let's ask @SamuelCahyawijaya to review LangMod task on this dataset

Comment (SamuelCahyawijaya) [reply seacrowd/sea_datasets/xstorycloze/xstorycloze.py:111]: @sabilmakbar  @akhdanfadh  @ljvmiranda921 : Hi all, sorry for the late reply. We can remove the Language Modeling task from the implementation. Sorry for the confusion 🙏🏻

**Comment by sabilmakbar** [comment]
Hi @akhdanfadh, have you got the time to check on this? Let us know if you need any help/more time to complete this PR; thanks!

**Comment by akhdanfadh** [comment]
`_ssp` schema implementation has been removed. Also added `_LANGUAGES`. Based on recent comments, all should be good now.

**Review by ljvmiranda921 — APPROVED** [review]
LGTM! Let's just wait for @sabilmakbar 's review :)

**Comment by sabilmakbar** [comment]
Welcome back, @akhdanfadh. I hope everything is well (since I heard you took some rest). I will check the new changes by EoD. I hope that all of our prior discussions have been taken care of.

And one more thing: Would you mind removing the `wongnai_reviews` changes from this PR? Thanks!

File: seacrowd/sea_datasets/xstorycloze/xstorycloze.py
Comment (sabilmakbar) [inline seacrowd/sea_datasets/xstorycloze/xstorycloze.py:2]: remove this as it's unused (from code formatted)
```suggestion
import itertools
```

File: seacrowd/sea_datasets/xstorycloze/xstorycloze.py
Comment (sabilmakbar) [inline seacrowd/sea_datasets/xstorycloze/xstorycloze.py:66]: use PascalCase (first letter is capitalized too)
```suggestion
class XStoryClozeDataset(datasets.GeneratorBasedBuilder):
```

**Review by sabilmakbar — CHANGES_REQUESTED** [review]
minor changes, else all good. will approve once these has been addressed (and the `wongnai_reviews` dataloader has been removed from this PR)

**Review by sabilmakbar — APPROVED** [review]
LGTM, thanks a lot, @akhdanfadh!

**Pull request merged**
